### PR TITLE
feat: vpc peering connection

### PR DESCRIPTION
**Pull request by fanriming**
#### What type of this PR
Examples of user facing changes:
- API changes

Bring VPC Peering feature to kube-ovn
https://docs.aws.amazon.com/vpc/latest/peering/what-is-vpc-peering.html

#### OVN command example
```
ovn-nbctl lr-add lr0
ovn-nbctl lr-add lr1
ovn-nbctl lrp-add lr0 lr0-p0 00:11:22:33:44:55 192.168.10.10/24 peer=lr1-p0
ovn-nbctl lrp-add lr1 lr1-p0 00:11:22:33:44:66 192.168.20.10/24 peer=lr0-p0
ovn-nbctl lrp-del lr1-p0
```
#### USAGE
Create two VPCs with static route and peering connection
```
kind: Vpc
apiVersion: kubeovn.io/v1
metadata:
  name: test-vpc-1
spec: 
  staticRoutes:
  - cidr: 10.0.2.0/24
    nextHopIP: 169.254.0.2
    policy: policyDst
  vpcPeerings:
  - localConnectIP: 169.254.0.1/30
    remoteVpc: test-vpc-2
---
kind: Subnet
apiVersion: kubeovn.io/v1
metadata:
  name: net1
spec:
  vpc: test-vpc-1
  cidrBlock: 10.0.1.0/24
  default: true
  natOutgoing: false
---
kind: Vpc
apiVersion: kubeovn.io/v1
metadata:
  name: test-vpc-2
spec: 
  staticRoutes:
  - cidr: 10.0.1.0/24
    nextHopIP: 169.254.0.1
    policy: policyDst
  vpcPeerings:
  - localConnectIP: 169.254.0.2/30
    remoteVpc: test-vpc-1
---
kind: Subnet
apiVersion: kubeovn.io/v1
metadata:
  name: net2
spec:
  vpc: test-vpc-2
  cidrBlock: 10.0.2.0/24
  natOutgoing: false
```
Traceroute result 
image

### CODE REVIEW

File: pkg/controller/gc.go [hunk 1]
```diff
@@ -39,6 +40,36 @@ func (c *Controller) gc() error {
...
+
+	var exceptPeerPorts []string
+	for _, vpc := range vpcs {
+		for _, peer := range vpc.Status.VpcPeerings {
+			exceptPeerPorts = append(exceptPeerPorts, fmt.Sprintf(vpc.Name, peer))
+		}
+	}
+	lrps, err := c.ovnClient.ListLogicalEntity("logical_router_port", "peer!=[]")
+	if err != nil {
+		klog.Errorf("failed to list logical switch port, %v", err)
```
Comment (oilbeater) [inline pkg/controller/gc.go:59]: `logical switch port` -> `logical router port`

File: pkg/controller/gc.go [hunk 2]
```diff
@@ -39,6 +40,36 @@ func (c *Controller) gc() error {
...
+	vpcs, err := c.vpcsLister.List(labels.Everything())
+	if err != nil {
+		klog.Errorf("failed to list vpc, %v", err)
+		return err
+	}
+
+	var exceptPeerPorts []string
+	for _, vpc := range vpcs {
+		for _, peer := range vpc.Status.VpcPeerings {
+			exceptPeerPorts = append(exceptPeerPorts, fmt.Sprintf(vpc.Name, peer))
```
Comment (oilbeater) [inline pkg/controller/gc.go:54]: `fmt.Sprintf("%s-%s")`

File: pkg/ovs/ovn-nbctl.go [hunk 3]
```diff
@@ -727,6 +716,55 @@ func (c Client) createRouterPort(ls, lr, ip, mac string) error {
 	return nil
 }
 
+func (c Client) GetLogicalRouterPortNetworks(port string) ([]string, error) {
```
Comment (oilbeater) [inline pkg/ovs/ovn-nbctl.go:719]: This function is not used

**Review by oilbeater — CHANGES_REQUESTED** [review]

**Comment by oilbeater** [comment]
@fanriming can you  add the docs?

**Comment by fanriming** [comment]
@oilbeater

File: pkg/controller/gc.go (hunk 1 above)
Comment (fanriming) [reply pkg/controller/gc.go]: fixed

File: pkg/controller/gc.go (hunk 2 above)
Comment (fanriming) [reply pkg/controller/gc.go]: fixed

File: pkg/ovs/ovn-nbctl.go (hunk 3 above)
Comment (fanriming) [reply pkg/ovs/ovn-nbctl.go]: fixed

**Comment by fanriming** [comment]
> @fanriming can you add the docs?
Docs has been added to commit now.

**Review by oilbeater — APPROVED** [review]

**Comment by xujunjie-cover** [comment]
Can it support visit peer vpc service ?

**Pull request merged**